### DISCUSSION TITLE
Is this awesome library support tanstack/react-query v4

### DISCUSSION THREAD

**Discussion by asifka-ramz** [KevinVandy/material-react-table#827] [Q&A]
Hi All,
Is this awesome library support tanstack/react-query v4 ?  and the document mentioned  like "make sure that the data is memoized or stable (i.e. useState, useMemo, defined outside of your table component, etc.). Otherwise you may get infinite renders."  but in examples using useQuery  for data fetching , we can see data fetching call inside the component  
<img width="547" alt="image" src="https://github.com/KevinVandy/material-react-table/assets/120662748/0c65e88b-9765-4cfa-8f52-a4afeedbde31">

but in my case when call data fetching inside component it makes browser stuck and high memory usage, actually i am using custom hook to fetch data using tanstack/ react-query v4.  

Is this support react-query v4?

how to use custom hook  for fetching data without browser stuck

Thanks

**Comment by KevinVandy** [comment]
This library doesn't care what version of TanStack Query you use. I just keep the examples up to date with the latest versions of other libraries. Don't just blindly copy and paste examples. Try to understand them and write your own code for your own needs with your other own libraries.

**Reply by asifka-ramz** [reply]
Thanks for reply

**Comment by KevinVandy** [comment]
useQuery provides data a stable reference like useState
